### PR TITLE
5 due date and priority functionality

### LINKED ISSUE #5: Due Date and Priority Functionality
## Value Statement
As a **user**
**I want** to set due dates and priorities for my todo items
**so** I can better manage my tasks.

## Description
Add due date and priority fields to todo items and implement sorting/filtering based on these fields.

## Acceptance criteria
- [x] Update todo item model to include due_date and priority fields
- [x] Ability to set and update due date and priority when creating or updating a todo item
- [x] GET /todos?sort_by={due_date, priority, created_at} endpoint to sort todos by due date, priority, or created_at. created_at should be the default.
- [x] GET /todos?due_date={some_date} endpoint to filter todos by due_date
- [x] GET /todos?priority={high/medium/low} endpoint to filter todos by priority
- [x] GET /todos?order={asc/desc} endpoint to arrange todos in ascending or descending order. Desc should be the default.

## Tasks
- [x] Update TodoItem SQLAlchemy model to include due_date and priority fields
- [x] Update TodoItem Pydantic schema with new fields
- [x] Modify POST and PUT /todos endpoints to handle new fields
- [x] Implement filtering logic for due_date and priority in GET /todos endpoint
- [x] Implement sorting logic for due_date, priority, and created_at in GET /todos endpoint
- [x] Implement ascending and descending logic in GET /todos endpoint
- [x] Update unit tests to cover new functionality
- [x] Update API documentation with new fields and sorting options

## Size
medium

### PR DESCRIPTION
**Pull request by jecaps**
- installed alembic 
- updated schemas and models
- implemented get_todos endpoint
- created todos_manager
- implemented sorting and filtering on todos
- updated the documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,9 +102,29 @@ The project includes two main models:
...
+        ```
     - Response: Created todo object
 
+
+- **GET /todos**: Retrieve todos with filtering and sorting
+    - Query Parameters:
+        - due_date: Filter by due_date
+        - priority: Filter by priority (high, medium, low)
+        - sort_by: Sort by field (due_date, priority, created_at)
+        - order: Sort order (asd, desc)
```
Comment (akanz1) [inline README.md]:
```suggestion
        - order: Sort order (asc, desc)
```

File: alembic/versions/1613c3341f78_add_due_date_and_priority_to_todos.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+revision: str = '1613c3341f78'
+down_revision: Union[str, None] = None
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    op.add_column('todos', sa.Column('due_date', sa.DateTime(), nullable=True))
+    op.add_column('todos', sa.Column('priority', sa.Enum('low', 'medium', 'high', name='priority'), nullable=False, server_default='medium'))
+    pass
```
Comment (akanz1) [inline alembic/versions/1613c3341f78_add_due_date_and_priority_to_todos.py]: no need for the pass statements here and in L.30

File: app/todo_manager.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
+from datetime import datetime
+from enum import Enum
+from typing import List
```
Comment (akanz1) [inline app/todo_manager.py]: use `list` instead of importing List from typing.

**Review by akanz1 — CHANGES_REQUESTED** [review]
great!

**Comment by jecaps** [comment]
@akanz1

**Review by akanz1 — APPROVED** [review]

**Pull request merged**